### PR TITLE
Add etcd robustness presubmit job

### PR DESCRIPTION
**Pull request by ArkaSaha30**
This commit will migrate etcd robustness [presubmit github action](https://github.com/etcd-io/etcd/blob/main/.github/workflows/robustness.yaml) to presubmit prowjob for `main` branch.

Issue link: https://github.com/etcd-io/etcd/issues/18136

### CODE REVIEW

**Comment by ArkaSaha30** [comment]
cc @ivanvc

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (serathius) [inline config/jobs/etcd/etcd-presubmits.yaml:241]: No eks for robustness. It's too slow.

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (ivanvc) [inline config/jobs/etcd/etcd-presubmits.yaml:250]: I think we can also add it to the robustness dashboard

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (ArkaSaha30) [reply config/jobs/etcd/etcd-presubmits.yaml]: Done 👍

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (ArkaSaha30) [reply config/jobs/etcd/etcd-presubmits.yaml]: Done 👍

Comment (ivanvc) [reply config/jobs/etcd/etcd-presubmits.yaml]: Sorry if I wasn't clear, but I meant to add rather than replace. It can appear in more than one dashboard.

**Comment by serathius** [comment]
/lgtm
/hold
cc @jmhbnz

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (jmhbnz) [inline config/jobs/etcd/etcd-presubmits.yaml:250]: Let's show this job on both dashboards as suggested above by @ivanvc.

```suggestion
      testgrid-dashboards: sig-etcd-presubmits, sig-etcd-robustness
```

**Review by jmhbnz — COMMENTED** [review]
Thanks @ArkaSaha30 looking good, just one tweak below before we merge.

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (ArkaSaha30) [reply config/jobs/etcd/etcd-presubmits.yaml]: Yeah, I misread that, adding it to both.

File: config/jobs/etcd/etcd-presubmits.yaml
Comment (ArkaSaha30) [reply config/jobs/etcd/etcd-presubmits.yaml:250]: Yupp, adding it to both. Thank you!

Comment (ArkaSaha30) [reply config/jobs/etcd/etcd-presubmits.yaml]: Done 👍

**Review by ivanvc — APPROVED** [review]
LGTM

**Review by jmhbnz — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
